test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -220,6 +227,7 @@
     private Function<Client, Client> clientWrapper;
 
     public InternalTestCluster(String nodeMode, long clusterSeed, Path baseDir,
+                               int numMasterNodes,
```

Comment:
Let's separate what we already have (my fault!) with what we are adding. I think either a test can work against dedicated masters or it can't, if that assumption is right then the api should reflect that, especially if it simplifies things. Why give the option to control how many dedicated master if not needed? I would go for the boolean here unless we really think we are going to need to set the specific number of nodes. With client nodes it might have been different in the beginning as there used to be tests that worked only against one single client node (remember the former shouldConnectTo method? :)  )